framework/src/modules/chain/chain.js

```diff
@@ -268,8 +269,32 @@ module.exports = class Chain {
 			}),
 			blocks: async action =>
 				this.scope.modules.transport.shared.blocks(action.params || {}),
-			blocksCommon: async action =>
-				this.scope.modules.transport.shared.blocksCommon(action.params || {}),
+
+			getHighestCommonBlockId: async action => {
+				const valid = this.scope.schema.validate(
```

Comment:
Would be nice if we wouldn't need all this logic inside the chain actions to keep this clean?

Reply:
We definitely need to think what to do with this, if Transport is a good way to hold action query schema validation and logic routing. I am personally not a big fan of the current state of transport.js as it doesn't only hold transport logic strictly.

From my perspective, Transport could hold action query schema validation and response formatting and in case the validation succeeds could route the query to the desired logic stored in another chain submodule, but as of now, this is not the case. We have to discuss this further.

Reply:
Isn't tranport doing the validation schema currently for all the other actions?